pkg/operator/configobservation/images/observe_images.go

```diff
@@ -98,13 +100,15 @@ func ObserveExternalRegistryHostnames(genericListers configobserver.Listers, exi
 	externalRegistryHostnames := configImage.Spec.ExternalRegistryHostnames
 	externalRegistryHostnames = append(externalRegistryHostnames, configImage.Status.ExternalRegistryHostnames...)
 
-	hostnames, err := Convert(externalRegistryHostnames)
-	if err != nil {
-		return prevObservedConfig, append(errs, err)
-	}
-	err = unstructured.SetNestedField(observedConfig, hostnames, externalRegistryHostnamePath...)
-	if err != nil {
-		return prevObservedConfig, append(errs, err)
+	if len(externalRegistryHostnames) > 0 {
```

Comment:
i don't think this is correct.  We have to be able to blank out the value if all the externalRegistryHostnames are removed.

Reply:
> i don't think this is correct. We have to be able to blank out the value if all the externalRegistryHostnames are removed.

`observedConfig` starts out as empty (missing entirely).  Wouldn't this code just keep that (value is entirely missing) when the list only has zero entries?

Reply:
hm. I had a misunderstanding of what is done w/ the returned values.